gax-java/gax/src/main/java/com/google/api/gax/rpc/TransportChannelProvider.java

```diff
@@ -88,15 +89,20 @@ public interface TransportChannelProvider {
   TransportChannelProvider withHeaders(Map<String, String> headers);
 
   /** True if the TransportProvider has no endpoint set. */
+  @ObsoleteApi("Use endpointContext.resolvedEndpoint() instead")
```

Comment:
```suggestion
  @ObsoleteApi("Endpoint should be configured set in the ClientSettings")
```

Reply:
Done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/2f70bf80b7f870fb426a73b6caecd6ea473c25d8